docs/usage/querying.rst

```diff
@@ -31,8 +31,16 @@ Print unique identification (Id) and last name of all employees:
 .. code-block:: python
 
     employees = northwind.entity_sets.Employees.get_entities().select('EmployeeID,LastName').execute()
-    for employee in employees:
-        print(employee.EmployeeID, employee.LastName)
+    while True:
+        for employee in employees:
+            print(employee.EmployeeID, employee.LastName)
+
+        # Stop if server has no more entities left
+        if employees.next_url is None:
+            break
+
+        # We got a partial answer - continue with next page
+        employees = northwind.entity_sets.Employees.get_entities().next_url(employees.next_url).execute()
```

Comment:
Oops, I didn't realize users will have to do this kind of exercise. Could you please create some Iterator wrapper? My goal was to create a library that integrates into python standard.

Reply:
@filak-sap I understand this does not look great. 

However, I would see also benefits in this approach. IMHO there is more control over the URLs sent, how it will look like (additional parameters) and how many requests are sent, when exactly and how big of response we receive. If hide subsequent requests behind iterator, that would yield entities based on presence of _next attribute, wouldn't we in the end transfer the entire DB to the memory in the model.py (for that particular entityset), regardless yielding of entity after entity out of the pyodata library to the client script? It is gut-feeling concern, I am not able to point to a code line that would do that.

Reply:
> However, I would see also benefits in this approach. IMHO there is more control over the URLs sent, how it will look like (possible additional parameters) and how many requests are sent, when exactly and how big of response we receive.

+1 - We definitely should give the user control over this.

> If hide subsequent requests behind iterator, that would yield entities based on presence of _next attribute, wouldn't we in the end transfer the entire DB to the memory in the model.py (for that particular entityset), regardless yielding of entity after entity out of the pyodata library to the client script?

Returning a generator which yields individual entities (instead of a single `ListWithTotalCount`) could easily be implemented to not issue a 2nd request towards the OData server until all available entities have been consumed. This keeps the memory consumption on the level we have currently.

To avoid API breakage, how about this: Add a method `recursive()` to class `QueryRequest`. It could take a parameter `strategy` which controls whether to return a list of class `ListWithTotalCount` or a generator which either returns a generator of `ListWithTotalCount` or entity objects. Default would be the "strategy" which is currently implemented: Leaving it up to the user.

Reply:
@rettichschnidi  Thank you for the proposal! Why not to make the comfortable strategy the default when you will add another method anyways? Could you please post some pseudo code?

Reply:
I'd have to either break the API (switch to a generator) or keep all elements in memory (so that I can assemble a ListWithTotalCount which contains truly all elements) and therefore risking some kind of resource exhaustion.

Tell me which you you prefer and I will post some code.

Reply:
OK, the task seems to be too complex, so let's merge the current version and perhaps add the helper in future.

Reply:
How about passing ListWithTotalCount? Something like this:

```python
employees = northwind.entity_sets.Employees.get_entities().next_page(employees).execute()
```

Reply:
IMHO throwing exception to in essence valid scenario is even worse than` if employees.next_url is None:` in a loop. Especially if the exception is related to odata protocol details.

**Option experimental**
We can take this in a way as an experimental API, (like they exists in Python 3.10.0 itself) - merge, not document, not in changelog, with follow-up issue that will either stick to it (and add documentation and release notes), change the API to some proposed in comments, or rollback the change.

Reply:
> We can take this in a way as an experimental API, (like they exists in Python 3.10.0 itself) - merge, not document, not in changelog, with follow-up issue that will either stick to it (and add documentation and release notes), change the API to some proposed in comments, or rollback the change.

I could roll with this. @phanak-sap: Any chance you'd accept this for version 1.8?

Reply:
I just pushed a new version without change note and no documentation change.

Reply:
side note: well, they can always checkout the current code from your fork for this PR to play with as I did. Does not need to be merged to master for that. And by the force-push 

Where this PR will be merged as currently is, or not, there are tasks outside this PR to track. I created issue #198 for that, so far just to link, will add task-list after this PR is resolved. 

@jfilak What is your preference here? Merge current code to master? Request Changes? This is PR that I am not comfortable to approve or disapprove just on my own.